app/controllers/ViewReturnSummaryController.scala

```diff
@@ -31,22 +36,47 @@ import scala.concurrent.{ExecutionContext, Future}
 class ViewReturnSummaryController @Inject() (
   override val messagesApi: MessagesApi,
   identify: IdentifierAction,
+  sessionRepository: SessionRepository,
+  getData: DataRetrievalAction,
   val controllerComponents: MessagesControllerComponents,
   view: ViewReturnSummaryView,
   taxReturnHelper: TaxReturnHelper
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController with I18nSupport {
 
   def onPageLoad(periodKey: String): Action[AnyContent] =
-    identify.async {
+    (identify andThen getData).async {
       implicit request =>
-        val enrolmentId = request.enrolmentId.getOrElse(throw new IllegalArgumentException("Make this not optional?")) // TODO
-        val submittedReturn: Future[ReturnDisplayApi] =
-          taxReturnHelper.fetchTaxReturn(enrolmentId, periodKey.toUpperCase())
-        submittedReturn.map {
-          val returnPeriod = "April to June 2022" // TODO
-          subRet => Ok(view(returnPeriod, ViewReturnSummaryViewModel(subRet)))
+        val enrolmentId = request.request.enrolmentId.getOrElse(throw new IllegalArgumentException("Make this not optional?")) // TODO
```

Comment:
Can we have a more appropriate message and perhaps a TODO to look at making this not optional?